client-spark/spark2/src/main/java/org/apache/spark/shuffle/DelegationRssShuffleManager.java

```diff
@@ -0,0 +1,183 @@
+        LOG.info("Use RssShuffleManager");
+        return shuffleManager;
+      } catch (Exception exception) {
+        LOG.warn("Fail to create RssShuffleManager, fallback to SortShuffleManager");
+      }
+    }
+
+    try {
+      shuffleManager = RssShuffleUtils.loadShuffleManager(sortShuffleManagerName, sparkConf, true);
+      sparkConf.set(RssClientConfig.RSS_USE_RSS_SHUFFLE_MANAGER, "false");
+      LOG.info("Use SortShuffleManager");
+    } catch (Exception e) {
+      throw new RssException(e.getMessage());
+    }
+
+    return shuffleManager;
+  }
+
+  private boolean tryAccessCluster() {
+    String accessId = sparkConf.get(
+        RssClientConfig.RSS_ACCESS_ID, RssClientConfig.RSS_ACCESS_ID_DEFAULT_VALUE).trim();
+    if (StringUtils.isEmpty(accessId)) {
+      LOG.warn("Access id key is empty");
+      return false;
+    }
+
+    for (CoordinatorClient coordinatorClient : coordinatorClients) {
+      try {
+        RssAccessClusterResponse response =
+            coordinatorClient.accessCluster(new RssAccessClusterRequest(
+                accessId, Sets.newHashSet(Constants.SHUFFLE_SERVER_VERSION), accessTimeoutMs));
+        if (response.getStatusCode() == ResponseStatusCode.SUCCESS) {
+          LOG.warn("Success to access cluster {} using {}", coordinatorClient.getDesc(), accessId);
+          return true;
+        } else if (response.getStatusCode() == ResponseStatusCode.ACCESS_DENIED) {
+          LOG.warn("Request to access cluster {} is denied using {} for {}",
+              coordinatorClient.getDesc(), accessId, response.getMessage());
+          return false;
+        } else {
+          LOG.warn("Fail to reach cluster {} for {}", coordinatorClient.getDesc(), response.getMessage());
+        }
+      } catch (Exception e) {
+        LOG.warn("Fail to access cluster {} using {} for {}",
+            coordinatorClient.getDesc(), accessId, e.getMessage());
+      }
+    }
+
+    return false;
+  }
+
+  private ShuffleManager createShuffleManagerInExecutor() throws RssException {
+    ShuffleManager shuffleManager;
+    // get useRSS from spark conf
+    boolean useRSS = Boolean.parseBoolean(sparkConf.get(
+        RssClientConfig.RSS_USE_RSS_SHUFFLE_MANAGER,
+        RssClientConfig.RSS_USE_RSS_SHUFFLE_MANAGER_DEFAULT_VALUE));
+    if (useRSS) {
+      // Executor will not do any fallback
+      shuffleManager = new RssShuffleManager(sparkConf, false);
+      LOG.info("Use RssShuffleManager");
+    } else {
+      try {
+        shuffleManager = RssShuffleUtils.loadShuffleManager(sortShuffleManagerName, sparkConf, false);
+        LOG.info("Use SortShuffleManager");
+      } catch (Exception e) {
+        throw new RssException(e.getMessage());
+      }
+    }
+    return shuffleManager;
+  }
+
+  public ShuffleManager getDelegate() {
+    return delegate;
+  }
+
+  @Override
+  public <K, V, C> ShuffleHandle registerShuffle(int shuffleId, int numMaps, ShuffleDependency<K, V, C> dependency) {
```

Comment:
@frankliee Could you see the compatibility of 3.0 and 2.3?

Reply:
mvn package -Pspark2.3
mvn package -Pspark3.0